packages/devextreme/js/__internal/ui/list/m_list.base.ts

```diff
@@ -605,31 +640,62 @@ export const ListBase = CollectionWidget.inherit({
   },
 
   _attachGroupCollapseEvent() {
-    const eventName = addNamespace(clickEventName, this.NAME);
-    const selector = `.${LIST_GROUP_HEADER_CLASS}`;
+    const { collapsibleGroups } = this.option();
+
+    const eventNameClick = addNamespace(clickEventName, this.NAME);
+    const headerSelector = `.${LIST_GROUP_HEADER_CLASS}`;
+
     const $element = this.$element();
-    const collapsibleGroups = this.option('collapsibleGroups');
 
     $element.toggleClass(LIST_COLLAPSIBLE_GROUPS_CLASS, collapsibleGroups);
 
-    eventsEngine.off($element, eventName, selector);
+    eventsEngine.off($element, eventNameClick, headerSelector);
+
     if (collapsibleGroups) {
-      eventsEngine.on($element, eventName, selector, (e) => {
-        this._createAction((e) => {
-          const $group = $(e.event.currentTarget).parent();
-          this._collapseGroupHandler($group);
-          if (this.option('focusStateEnabled')) {
-            this.option('focusedElement', getPublicElement($group.find(`.${LIST_ITEM_CLASS}`).eq(0)));
-          }
-        }, {
-          validatingTargetName: 'element',
-        })({
-          event: e,
-        });
+      eventsEngine.on($element, eventNameClick, headerSelector, (e) => {
+        this._processGroupCollapse(e);
       });
     }
   },
 
+  _processGroupCollapse(e): void {
+    const actionCallback = (e) => {
+      const { focusStateEnabled } = this.option();
+      const $group = $(e.event.currentTarget).parent();
+
+      this._collapseGroupHandler($group);
+
+      if (focusStateEnabled) {
+        const listItemElement = getPublicElement($group.find(`.${LIST_ITEM_CLASS}`).eq(0));
+
+        this.option({ focusedElement: listItemElement });
+      }
+    };
+
+    const actionParams = {
+      validatingTargetName: 'element',
+    };
+
+    const action = this._createAction(actionCallback, actionParams);
+
+    action({ event: e });
+  },
+
+  _enterKeyHandler(e): void {
+    const { collapsibleGroups, focusedElement } = this.option();
+    const isGroupHeader = $(focusedElement).hasClass(LIST_GROUP_HEADER_CLASS);
```

Comment:
The mechanism by which focusing works in collection_widget is a bit different. Let me explain. document.activeElement is scrollview in this case, so when you press Enter in e.target it will be the scrollview and not the header. We store the focused element in the focusedElement option.

See how it works in collection_widget: 

https://github.com/DevExpress/DevExtreme/blob/24_2/packages/devextreme/js/__internal/ui/collection/m_collection_widget.base.ts#L89

Reply:
Oh, okay, I see